### PR TITLE
Add aten::slice NNAPI converter

### PR DESCRIPTION
**Pull request by axitkhurana**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #61025
* #61024
* #61023
* #61022
* #61021
* #59529
* #59434
* #59365
* **#59364**
* #58543
* #60885
* #58541
* #58540
* #58539

Summary:

Add support for aten::slice op in the NNAPI model converter

* If start = 0; end = max -> identity
* Flexible shapes can be passed through
* Flexible shapes can't be sliced over

Test Plan:

pytest test/test_nnapi.py::TestNNAPI::test_slice

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28881039](https://our.internmc.facebook.com/intern/diff/D28881039)

### CODE REVIEW

**Comment by axitkhurana** [comment]
@axitkhurana has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28881039).

**Comment by axitkhurana** [comment]
@axitkhurana has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28881039).

**Comment by axitkhurana** [comment]
@axitkhurana has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28881039).

**Comment by axitkhurana** [comment]
@axitkhurana has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28881039).

**Comment by axitkhurana** [comment]
@axitkhurana has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28881039).

**Comment by axitkhurana** [comment]
@axitkhurana has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28881039).

File: torch/backends/_nnapi/serializer.py [hunk 1]
```diff
@@ -971,6 +974,69 @@ def add_flatten(self, node):
...
+        _, step_value = self.get_constant_value(node.inputsAt(4))
+
+        if start_value is None:
+            start_value = 0
+        if stop_value is None:
+            stop_value = sys.maxsize
+
+        if start_value < 0:
+            start_value += in_oper.shape[dim_value]
+        elif start_value == sys.maxsize:
```
Comment (dreiss) [inline torch/backends/_nnapi/serializer.py:994]: Why would this happen?

**Review by dreiss — APPROVED** [review]

File: torch/backends/_nnapi/serializer.py (hunk 1 above)
Comment (axitkhurana) [reply torch/backends/_nnapi/serializer.py:994]: * https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/TensorIndexing.h#L18
* https://github.com/pytorch/pytorch/blob/master/test/cpp/api/tensor_indexing.cpp#L45

None ends up translating to INDEX_MAX

**Comment by axitkhurana** [comment]
@axitkhurana has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28881039).

**Pull request closed without merging**